test/Test/Util/FS.hs

```diff
@@ -368,6 +408,27 @@ instance Arbitrary NoCleanupErrors where
   Arbitrary: orphans
 -------------------------------------------------------------------------------}
 
+instance Arbitrary FsPath where
+  arbitrary = scale (`div` 10) $ -- paths don't have to be very long
+      FS.mkFsPath <$> listOf (fsPathComponentString <$> arbitrary)
+  shrink p =
+      let ss = T.unpack <$> fsPathToList p
+      in  FS.mkFsPath <$> shrinkList shrinkAsComponent ss
+    where
+      shrinkAsComponent s = fsPathComponentString <$>
+          shrink (FsPathComponent $ NE.fromList s)
+
+-- >>> [ c | c <- [minBound..maxBound], isPathChar c ]
+-- "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+isPathChar :: Char -> Bool
+isPathChar c = isAscii c && (isLetter c || isDigit c)
+
+genPathChar :: Gen Char
+genPathChar = arbitraryASCIIChar `suchThat` isPathChar
```

Comment:
Is this generating and filtering rather than generating from a list of options? Wouldn't that have really poor performance when generating?

Reply:
The performance is fine, but I agree it can be faster. I've changed it to pick `Char`s from a fixed list instead.